plugins/inputs/socket_listener/socket_listener.go

```diff
@@ -47,6 +48,27 @@ func (ssl *streamSocketListener) listen() {
 		}
 		ssl.connections[c.RemoteAddr().String()] = c
 		ssl.connectionsMtx.Unlock()
+
+		if ssl.KeepAlivePeriod != nil {
+			if tcpc, ok := c.(*net.TCPConn); ok {
+				if ssl.KeepAlivePeriod.Duration == 0 {
+					if err := tcpc.SetKeepAlive(false); err != nil {
+						ssl.AddError(fmt.Errorf("unable to disable keep alive: %s", err))
+					}
+				} else {
+					if err := tcpc.SetKeepAlive(true); err != nil {
+						ssl.AddError(fmt.Errorf("unable to enable keep alive: %s", err))
+					} else {
+						if err := tcpc.SetKeepAlivePeriod(ssl.KeepAlivePeriod.Duration); err != nil {
+							ssl.AddError(fmt.Errorf("unable to set keep alive period: %s", err))
```

Comment:
so
&nbsp;&nbsp;much
&nbsp;&nbsp;&nbsp;&nbsp;nesting

The only way I can think of to flatten this out a bit would be to break the code out into a function and use `return`s. Not sure how I feel about that.